### DISCUSSION TITLE
使用Cloudflare Workers代理时的端口问题

### DISCUSSION THREAD

**Discussion by jwan9714** [AlistGo/alist#3849] [Q&A]
请问各位大佬，使用Cloudflare Workers进行代理时Alist的端口http只能是80端口吗？

我看到[帮助文档](https://alist.nn.ci/zh/guide/drivers/common.html#%E4%B8%8B%E8%BD%BD%E4%BB%A3%E7%90%86-url)中提到的是国内，我在国外的服务器搭建并使用国外代理进行下载时还是会报错。

Alist端口为8080，workers脚本如下：

```
var ADDRESS = "http://alist.xxx.com:8080";
var TOKEN = "alist-xxxxx
```

cloudflare报错为Error 1101，日志如下：

```
{
  "outcome": "exception",
  "scriptName": "alist",
  "exceptions": [
    {
      "name": "SyntaxError",
      "message": "Unexpected end of JSON input",
      "timestamp": 1678792073108
    }
  ],
```

Alist使用80端口时，workers地址部分不填端口可以正常下载

**Comment by liueic** [comment]
同问😭

**Comment by wuzhihui04** [comment]
cloudflare的代理只放行80和443端口。

**Comment by sd14** [comment]
我试过5244端口都能通，你们怎么弄的。。
个人认为 var ADDRESS = "http://alist.xxx.com:8080";
这里填的 http://alist.xxx.com:8080 只要是外网能访问到alist就可以了

**Reply by liueic** [reply]
我用宝塔做了个反代，直接用域名访问就可以进入alist，配置文件里面也改为80端口了，但是使用cf worker的时候还是会出现ERROR 1101的情况

**Reply by sd14** [reply]
我刚去测试了，没有问题，alist版本3.30，端口5244，没有做反代，域名+端口访问alist
可能是反代导致哪里出错

**Comment by imHansiy** [comment]
<img width="249" alt="image" src="https://github.com/alist-org/alist/assets/48616740/b0753c34-ee91-46b5-b479-618aa8ff1e33">
<img width="215" alt="image" src="https://github.com/alist-org/alist/assets/48616740/eec7efe7-9a8d-4a9b-bacf-5492a9c563ba">
<img width="341" alt="image" src="https://github.com/alist-org/alist/assets/48616740/4fe02550-8fd0-453f-992a-36e8ab0d8b89">

**Comment by hahach** [comment]
如果你的域名是CF上解析的，可以任意端口。
做法就是cname解析一个域名a到你的alist服务的域名。例如：
你CF上管理的域名叫 abc.com, alist在  abc.com:8080(任意域名都行，alist服务的域名不需要CF管理的)。
在CF上给一个cname解析 例如  list.abc.com 指向 abc.com
然后CF上设置Origin Rules.
规则为  传入 主机名 等于 list.abc.com时，重写到8080端口。

最后你就可以在CF worker里写 list.abc.com了。

这个方法适用于任何CF服务限定端口的绕过

**Comment by wuzhihui04** [comment]
非常感谢。学习到了。

hahach ***@***.***> 于2024年5月6日周一 18:04写道：
